File: aptoide-database/src/main/java/cm/aptoide/pt/database/room/FileToDownloadTypeConverter.java
package cm.aptoide.pt.database.room;

import androidx.room.TypeConverter;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.util.List;

public class FileToDownloadTypeConverter {

  @TypeConverter
  public static List<RoomFileToDownload> restoreFileToDownloadList(String listOfString) {
    return new Gson().fromJson(listOfString, new TypeToken<List<RoomFileToDownload>>() {
    }.getType());
  }

  @TypeConverter
  public static String saveFileToDownloadList(List<RoomFileToDownload> listOfString) {
    return new Gson().toJson(listOfString);
  }
}

File: downloads_database/src/main/java/cm/aptoide/pt/downloads_database/data/database/model/FileToDownloadTypeConverter.kt
package cm.aptoide.pt.downloads_database.data.database.model

import androidx.room.TypeConverter
import cm.aptoide.pt.downloads_database.data.database.model.FileToDownload
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

object FileToDownloadTypeConverter {
  @TypeConverter
  fun restoreFileToDownloadList(listOfString: String?): List<FileToDownload> {
    return Gson().fromJson(listOfString, object : TypeToken<List<FileToDownload?>?>() {}.type)
  }

  @TypeConverter
  fun saveFileToDownloadList(listOfString: List<FileToDownload?>?): String {
    return Gson().toJson(listOfString)
  }
}